### PR TITLE
cmake: add M_TUNE

### PR DESCRIPTION
**Pull request by Andarwinux**

### CODE REVIEW

**Comment by shinchiro** [comment]
This doesnt make sense..I would prefer expose `M_TUNE` var on configure rather than add logic there

**Comment by Andarwinux** [comment]
Okay, I'll add it later.

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,6 +34,12 @@ elseif(TARGET_CPU STREQUAL "aarch64")
     set(GCC_ARCH "armv8.2-a" CACHE STRING "Target architecture for aarch64")
 endif()
 
+if(GCC_ARCH STREQUAL "native")
+    set(M_TUNE "native" CACHE STRING "-mtune=native")
+else()
+    set(M_TUNE "generic" CACHE STRING "-mune=generic")
```
Comment (shinchiro) [inline CMakeLists.txt]: Just `set(M_TUNE "generic" CACHE STRING "Tunes the generated code for the specified cpu-type")` without if-else is enough. People who want to set it to `native` or different should know better

**Comment by Andarwinux** [comment]
This seems pointless, it is already possible to override the default -mtune with CLANG_FLAGS. so close this PR.

**Pull request closed without merging**